python-yaml: update source and build for python3

dangowrt opened the pull request:
Maintainer: @psycho-nico 
Compile tested: x86/64
Run tested: x86/64

dangowrt commented:
@commodo As you seem to be experienced with python stuff (I'm not), please take a look at this change as well.

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,62 @@
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://pyyaml.org/download/pyyaml/
-PKG_MD5SUM:=f50e08ef0fe55178479d3a618efe21db
+PKG_MD5SUM:=4c129761b661d181ebf7ff4eb2d79950
```
note: as far as i understand it, the MD5SUM can be discarded, since the PKG_HASH field will be preferred
no harm in leaving it 

i started removing them ; 
since these packages are intended to be used with OpenWrt/LEDE trunk, they should both handle the PKG_HASH field

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,62 @@
+define Package/python-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python
   DEPENDS:=+python +libyaml
```
you could move the `DEPENDS:=+libyaml` to the  `Package/python-yaml/Default` rule
and do  `DEPENDS+=+PACKAGE_python-yaml:python` here and `DEPENDS+=+PACKAGE_python3-yaml:python3`  to the Python3 variant

also, I encourage the use of the `+PACKAGE_python-yaml:python` & `+PACKAGE_python3-yaml:python3` style variables ;
i noticed, that if you just specify `+python` & `+python3` dependencies, both Python & Python3 interpreters get built, even tho, one gets installed on the target [ or packaged ]
when both interpreters get built, that also means that the host Python & Python3 interpreters get built, so the build time reduction can be significant

this is a general thing with the LEDE/OpenWrt build system, and maybe could be addressed in core ;
but, until then, to speed up builds, this is one way to do it

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,62 @@
+  VARIANT:=python3
+endef
+
+define PyBuild/Compile/Default
```
If you agree with this change https://github.com/openwrt/packages/pull/4558
you could switch this to use  `PyBuild/Compile` & `Py3Build/Compile` 
not sure why I didn't do these names in the first place

that change would allow mechanisms like
```
define PyBuild/Compile
< do extra pre-stuff >
$(call PyBuild/Compile/Default)
< do extra post-stuff >
```

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,62 @@
 PKG_LICENSE_FILES:=COPYING
 PKG_MAINTAINER:=Nicolas Thill <nico@openwrt.org>
 
 PKG_BUILD_DEPENDS:=python
```
you can remove this ; if you're building the `python3` variant, you'll also get the Python interpreter(s) built

commodo commented:
i thought the build system required/enforced the `Package/python-yaml/description` rules [maybe this changed] ;
if it builds and the build system does not complain about it missing, then I'm fine with not having

also, FWIW, @psycho-nico seems a bit inactive [ no offense intended ];
so, if he times out for a week, I'd say to merge this ;
[long term] regarding ownership of this package, you can take it, or i can, both work for me ; i don't seem to see about any fixed guidelines on this

rest looks fine :)

dangowrt commented:
@commodo I removed the MD5SUM and changed the dependencies as you suggested. I do agree with the changes to be made in https://github.com/openwrt/packages/pull/4558 , however, I'd rather see a batch-change of all existing packages once the change has been prepared.
I'd say with give @psycho-nico a bit more time till after the weekend and then get this merged.

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,60 @@
+define Package/python3-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python3
+  DEPENDS+=+PACKAGE_python-yaml:python3-base
```
this should be `PACKAGE_python3-yaml`
i also do this goof sometimes ;

commodo commented:
regarding my changes in PR #4558 : the only package that defined the `PyBuild/Compile/Default` rule, was the `python-lxml` ; and that's included in that PR ;
that PR was merged [just recently], so, changing the `PyBuild/Compile/Default` to `PyBuild/Compile` is recommended now

dangowrt commented:
@commodo fixed both now

commodo commented:
you were quick;
i was hoping i'd have time to test the  `python-base` vs  `python`  changes

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,59 @@
+define Package/python-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python
+  DEPENDS+=+PACKAGE_python-yaml:python-base
```
this also needs `+PACKAGE_python-yaml:python-codecs` so that `import yaml` works

commodo replied:
to add to this real quick [and mostly for reference] ; there's a difference about how Python & Python3 handle unicode
Python3 requires the codecs to be available at startup, while Python does not
they do take a bit of space, so [since they're not needed in Python] I packaged them separately

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,59 @@
+define Package/python3-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python3
+  DEPENDS+=+PACKAGE_python3-yaml:python3-base
```
this needs to be `+PACKAGE_python3-yaml:python3-light` to run
seems that the `python3-base` does not package the `re` package [while the `python-base` package does 
and python-yaml seems to need it

i tried just to add the `re` module into the `python3-base` package, but it seems to require the `enum` module
```
root@LEDE:/# python3
Python 3.6.1 (default, Jul  9 2017, 19:44:34) 
[GCC 5.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 8, in <module>
  File "/loader.py", line 4, in <module>
  File "/reader.py", line 22, in <module>
  File "/re.py", line 122, in <module>
ModuleNotFoundError: No module named 'enum'
>>> 
```

I stopped after the `enum` module was required.
Even tho it's a bit inconsistent with the `python` variant, I'd say to use the `python3-light` here as a dependency.
It seems to work fine with that [ re & enum modules are in there ].

commodo commented:
rest is all good
thanks :)

dangowrt commented:
@commodo I now use python-light and python3-light to depend upon. Please give it one last look and (hopefully) give your approval. I'll then merge it myself as @psycho-nico hasn't shown up till now.

commodo commented on lang/python/python-yaml/Makefile:
```diff
@@ -8,36 +8,59 @@
+define Package/python-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python
+  DEPENDS+=+PACKAGE_python-yaml:python-light
```
sorry if i was not accurate here
but `+PACKAGE_python-yaml:python-codecs`  is required :)

dangowrt replied:
@commodo Oops, I missed that. Fixed now.

commodo commented:
rest is good to go :)

commodo commented:
ack
good to go

thanks :)
